agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/LogProcessor.java

```diff
@@ -0,0 +1,118 @@
+
+import com.microsoft.applicationinsights.agent.internal.wasbootstrap.configuration.Configuration.ProcessorConfig;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static com.microsoft.applicationinsights.agent.internal.processors.ProcessorUtil.*;
+
+
+public class LogProcessor extends AgentProcessor {
+    private final List<AttributeKey<?>> fromAttributes;
+    private final List<Pattern> toAttributeRulePatterns;
+    private final List<List<String>> groupNames;
+    private final String separator;
+
+    public LogProcessor(@Nullable IncludeExclude include,
+                        @Nullable IncludeExclude exclude,
+                        List<AttributeKey<?>> fromAttributes,
+                        List<Pattern> toAttributeRulePatterns,
+                        List<List<String>> groupNames,
+                        String separator) {
+        super(include, exclude);
+        this.fromAttributes = fromAttributes;
+        this.toAttributeRulePatterns = toAttributeRulePatterns;
+        this.groupNames = groupNames;
+        this.separator = separator;
+    }
+
+    public static LogProcessor create(ProcessorConfig config) {
+        IncludeExclude normalizedInclude = config.include != null ? getNormalizedIncludeExclude(config.include) : null;
+        IncludeExclude normalizedExclude = config.exclude != null ? getNormalizedIncludeExclude(config.exclude) : null;
+        List<AttributeKey<?>> fromAttributes = new ArrayList<>();
+        if (config.body.fromAttributes != null) {
+            for (String attribute : config.body.fromAttributes) {
+                fromAttributes.add(AttributeKey.stringKey(attribute));
+            }
+        }
+        List<String> toAttributeRules = new ArrayList<>();
+        if (config.body.toAttributes != null) {
+            toAttributeRules.addAll(config.body.toAttributes.rules);
+        }
+        List<Pattern> toAttributeRulePatterns = new ArrayList<>();
+        if (config.body.toAttributes != null) {
+            for (String rule : config.body.toAttributes.rules) {
+                toAttributeRulePatterns.add(Pattern.compile(rule));
+            }
+        }
+        List<List<String>> groupNames = getGroupNamesList(toAttributeRules);
+        String separator = config.body.separator != null ? config.body.separator : "";
+        return new LogProcessor(normalizedInclude, normalizedExclude,
+                fromAttributes, toAttributeRulePatterns, groupNames, separator);
+    }
+
+    //fromAttributes represents the attribute keys to pull the values from to generate the new span name.
+    public SpanData processFromAttributes(SpanData log) {
```

Comment:
probably could have avoided some of the copy paste here by extracting out another common method into ProcessorUtil, but let's do that later in follow-up PR